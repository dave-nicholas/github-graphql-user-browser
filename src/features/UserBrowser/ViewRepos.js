import React from 'react';
import PropTypes from 'prop-types';

const ViewRepos = ({ repositories: { edges } }) =>
  !!edges.length ? (
    <ul className="collection">
      {edges.map(({ node: { url, name, description } }, i) => (
        <li className="collection-item" key={i}>
          <span className="title">
            <a href={url}>{name}</a>
          </span>
          <p>{description}</p>
        </li>
      ))}
    </ul>
  ) : (
    'No repos...'
  );

ViewRepos.propTypes = {
  repositories: PropTypes.shape({
    edges: PropTypes.arrayOf(
      PropTypes.shape({
        url: PropTypes.string,
        name: PropTypes.string,
        description: PropTypes.string
      })
    )
  }).isRequired
};

export default ViewRepos;
