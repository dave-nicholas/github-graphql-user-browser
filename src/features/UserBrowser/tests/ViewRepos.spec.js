import React from 'react';
import { shallow } from 'enzyme';
import ViewRepos from '../ViewRepos';
import toJson from 'enzyme-to-json';

describe('ViewRepos', () => {
  const wrapper = shallow(
    <ViewRepos
      repositories={{
        edges: [
          {
            node: {
              description: 'this is a test repo',
              name: 'test',
              url: 'http://test.com/test'
            }
          }
        ]
      }}
    />
  );

  it('should render link to a repo', () => {
    expect(wrapper.contains(<a href="http://test.com/test">test</a>)).toBe(
      true
    );
  });

  it('should render a repo description', () => {
    expect(wrapper.text().indexOf('this is a test repo')).toBeGreaterThan(-1);
  });

  it('Should handle no repos', () => {
    expect(
      shallow(
        <ViewRepos
          repositories={{
            edges: []
          }}
        />
      )
        .text()
        .indexOf('No repos...')
    ).toBeGreaterThan(-1);
  });

  it('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
