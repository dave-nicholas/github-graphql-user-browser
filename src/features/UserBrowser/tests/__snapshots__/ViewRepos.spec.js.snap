// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRepos should match snapshot 1`] = `
<ul
  className="collection"
>
  <li
    className="collection-item"
    key="0"
  >
    <span
      className="title"
    >
      <a
        href="http://test.com/test"
      >
        test
      </a>
    </span>
    <p>
      this is a test repo
    </p>
  </li>
</ul>
`;
